src/App.tsx

```diff
@@ -1,223 +1,291 @@
+        }
+      } else {
+        setErrorType(ErrorType.empty);
+        setHasError(true);
+      }
+    }, [query, user],
+  );
 
-            <span data-cy="TodoTitle" className="todo__title">CSS</span>
+  const clearCompleted = async () => {
+    try {
+      setTodos(currTodos => currTodos.map(todo => (
+        todo.completed
+          ? ({ ...todo, isLoading: true })
+          : todo)));
 
-            <button
-              type="button"
-              className="todo__remove"
-              data-cy="TodoDeleteButton"
-            >
-              ×
-            </button>
-
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
-
-            <form>
-              <input
-                data-cy="TodoTitleField"
-                type="text"
-                className="todo__title-field"
-                placeholder="Empty todo will be deleted"
-                defaultValue="JS"
-              />
-            </form>
-
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
+      await Promise.all(complitedTodos.map(todo => (
+        deleteTodos(todo.id)
+      )));
+
+      loadTodos();
+    } catch {
+      setHasError(true);
+      setErrorType(ErrorType.delete);
+    }
+  };
+
+  const onRemoveTodo = async (id: number) => {
+    setTodos(currTodos => currTodos.map(todo => (
+      todo.id === id
+        ? ({ ...todo, isLoading: true })
+        : todo)));
+
+    try {
+      await deleteTodos(id);
+
+      loadTodos();
+    } catch {
+      setHasError(true);
+      setErrorType(ErrorType.delete);
+    } finally {
+      setHasError(false);
+    }
+  };
+
+  const onToggleTodo = async (id: number, completed: boolean) => {
+    setTodos(currTodos => currTodos.map(todo => (
+      todo.id === id
+        ? ({ ...todo, isLoading: true })
+        : todo)));
 
-            <span data-cy="TodoTitle" className="todo__title">React</span>
+    try {
+      await updateTodos(id, { completed: !completed });
+
+      loadTodos();
+    } catch {
+      setHasError(true);
+      setErrorType(ErrorType.update);
+    } finally {
+      setHasError(false);
+    }
```

Comment:
`Finally` works in any case even if there is an error. So in this case, even if you catch an error, change the boolean value to no error. Fix it